zfsbootmenu/lib/zfsbootmenu-lib.sh

```diff
@@ -251,7 +254,34 @@ draw_snapshots() {
 ^[CTRL+D] show diff
 ^[CTRL+H] help" )"
 
-  context="Note: for diff viewer, use tab to select/deselect up to two items"
+  sort_key="$( get_sort_key )"
+
+  sorted_by="Sorted by: ${sort_key^}"
+  note="Note: for diff viewer, use tab to select/deselect up to two items"
+
+  local LEGACY_CONTEXT
+  if [ -n "${HAS_BORDER}" ] ; then
+
+    # Determine how much space should be between the 'sorted by' text and a centered note
+    # Remove 4 extra characters so that we can put a 1 character pad between strings and 
+    # the horizontal box line
+
+    spacer=$(( ( ( COLUMNS - ${#note} ) / 2 ) - ${#sorted_by} - 4 ))
```

Comment:
Logic suggestion:

1. If `spacer` (should this be `local`?) is positive, print both sorted key and note as is done here.
2. Otherwise, if `$COLUMNS` is larger than the length of the note (plus some pad), just print the note.
3. Otherwise, if `$COLUMNS` is larger than the length of the sort key, just print the sort key.
4. Otherwise, omit context altogether.

Reply:
I've implemented that. I had thought of handling small screen sizes and then I got distracted.